# 2주차 정답 Notebook — 파이썬과 데이터 분석의 기본

## 1단계. 온도 변수 만들기

In [1]:
temp1 = 302.7
print(temp1)
print(type(temp1))

302.7
<class 'float'>


## 2단계. 리스트에 여러 온도 저장하기

In [2]:
temps = [302.7, 300.1, 292.0, 299.0, 291.9, 303.5]
print(temps)
print("측정 개수:", len(temps))

[302.7, 300.1, 292.0, 299.0, 291.9, 303.5]
측정 개수: 6


## 3단계. 평균/최댓값/최솟값 계산하기

In [3]:
average = sum(temps) / len(temps)
maximum = max(temps)
minimum = min(temps)

print("평균:", round(average, 1), "최댓값:", maximum, "최솟값:", minimum)

평균: 298.2 최댓값: 303.5 최솟값: 291.9


**질문 답**: 평균은 약 298.2℃, 최댓값은 303.5℃, 최솟값은 291.9℃이다.

## 4단계. 정상범위(295~305℃) 조건 확인하기

In [4]:
if maximum > 305 or minimum < 295:
    print("정상범위를 벗어난 값이 있습니다.")
else:
    print("모든 값이 정상범위 안에 있습니다.")

정상범위를 벗어난 값이 있습니다.


## 5단계. 전체 데이터를 Pandas로 불러와 리스트로 바꾸기

In [5]:
import pandas as pd

df = pd.read_csv("../../data/weekly/week02/week02_basic_process_values.csv")
df.head()

,측정시간,로트번호,공정명,온도_섭씨,압력_Pa,합격여부
0,2024-03-01 10:00,LOT-0139,증착,302.7,1005.9,1
1,2024-03-01 16:00,LOT-0088,식각,300.1,1001.3,1
2,2024-03-01 21:00,LOT-0045,증착,292.0,998.9,1
3,2024-03-02 04:00,LOT-0170,포토,299.0,1021.4,1
4,2024-03-02 07:00,LOT-0138,식각,291.9,1011.8,1


In [6]:
all_temps = df["온도_섭씨"].tolist()

print("전체 평균:", round(sum(all_temps) / len(all_temps), 1))
print("전체 최댓값:", max(all_temps), " 전체 최솟값:", min(all_temps))

전체 평균: 301.9
전체 최댓값: 320.0  전체 최솟값: 289.3


**질문 답**: 6개짜리 리스트(평균 298.2℃)와 전체 25건(평균 301.9℃)의 결과가 다르다.
표본 개수가 적으면 극단값(320.0℃, 289.3℃ 등)이 포함되지 않아 전체 상황을 대표하지 못하기 때문이다.

## 6단계. 수율(합격 비율) 계산하기

In [7]:
pass_count = len(df[df["합격여부"] == 1])
total_count = len(df)
yield_pct = pass_count / total_count * 100

print(f"수율: {pass_count}/{total_count} = {round(yield_pct, 1)}%")

수율: 23/25 = 92.0%


## 7단계(도전). 압력 리스트와 합격/불합격 그룹 비교하기

In [8]:
pressures = df["압력_Pa"].tolist()

print("압력 평균:", round(sum(pressures) / len(pressures), 1))
print("압력 최댓값:", max(pressures), " 압력 최솟값:", min(pressures))

압력 평균: 1010.6
압력 최댓값: 1027.7  압력 최솟값: 992.3


In [9]:
pass_avg_temp = df[df["합격여부"] == 1]["온도_섭씨"].mean()
fail_avg_temp = df[df["합격여부"] == -1]["온도_섭씨"].mean()

print("합격 그룹 평균 온도:", round(pass_avg_temp, 1))
print("불합격 그룹 평균 온도:", round(fail_avg_temp, 1))

합격 그룹 평균 온도: 302.2
불합격 그룹 평균 온도: 297.6


## 8단계. 오늘의 분석을 한 문장으로 정리하기

> 이번 배치 25건의 평균 온도는 301.9℃였고, 최댓값은 320.0℃까지 올라간 경우도 있었다.
> 전체 수율은 92.0%(23/25건 합격)였다. 흥미롭게도 최댓값이 나온 로트는 합격 처리되었고,
> 실제 불합격 2건은 오히려 완만하게 정상범위를 벗어난 값에서 발생했다.